src/Commands/AdminCommands/SendtochannelCommand.php

```diff
@@ -24,46 +24,253 @@ class SendtochannelCommand extends AdminCommand
     protected $name = 'sendtochannel';
     protected $description = 'Send message to a channel';
     protected $usage = '/sendtochannel <message to send>';
-    protected $version = '0.1.1';
-    protected $need_mysql = false;
+    protected $version = '0.1.2';
+    protected $need_mysql = true;
     /**#@-*/
 
     /**
-     * Execute command
-     *
-     * @todo Don't use empty, as a string of '0' is regarded to be empty
-     *
-     * @return boolean
+     * {@inheritdoc}
      */
     public function execute()
     {
+        $update = $this->getUpdate();
         $message = $this->getMessage();
 
-        $chat_id = $message->getChat()->getId();
+        $chat = $message->getChat();
+        $user = $message->getFrom();
+        $type = $message->getType();
+
+        $chat_id = $chat->getId();
+        $text = trim($message->getText(true));
+        $user_id = $user->getId();
+
         $text = $message->getText(true);
 
-        if (empty($text)) {
-            $text_back = 'Write the message to send: /sendtochannel <message>';
+        $data = [];
+        $data['chat_id'] = $chat_id;
+
+        //tracking
+        $conversation = new Conversation($user_id, $chat_id, $this->getName());
+        $conversation->start();
+        $session = $conversation->getData();
+
+        if (!isset($session['state'])) {
+            $state = '0';
+            $session['last_message_id'] = $message->getMessageId();
         } else {
-            $your_channel = $this->getConfig('your_channel');
-            //Send message to channel
-            $data = [
-                'chat_id' => $your_channel,
-                'text'    => $text,
-            ];
-
-            if (Request::sendMessage($data)->isOk()) {
-                $text_back = 'Message sent succesfully to: ' . $your_channel;
-            } else {
-                $text_back = 'Sorry message not sent to: ' . $your_channel;
-            }
+            $state = $session['state'];
+        }
+
+        $channels = $this->getConfig('your_channel');
```

Comment:
Before continuing, make sure that at least one channel has been set in the config.